### DISCUSSION TITLE
[Feature] [Security] Ensure immich containers work with readOnlyRootFilesystem enabled

### DISCUSSION THREAD

**Discussion by ItalyPaleAle** [immich-app/immich#4533] [Feature Request]
### The feature

[`readOnlyRootFilesystem`](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) (K8s) / `--read-only=true` ([Docker CLI](https://docs.docker.com/engine/reference/commandline/run/)) / `read_only: true` ([Docker Compose](https://github.com/compose-spec/compose-spec/blob/master/spec.md#read_only)) is an important security feature that makes the root filesystem of the container read-only. This means that in case of compromise, an attacker would not be able to modify the application's code.

Based on my own testing, the following containers seem to be working with a read-only root file-system:

- immich-server
  - Both server and microservices
  - Assuming a volume is mounted for the library and geodata cache
- immich-web
- immich-machine-learning
  - Assuming a volume is mounted for the library and for the ML model cache

These containers do not start up if the root FS is read-only:

- immich-proxy

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by etnoy** [comment]
Now that the proxy container is removed, please try again
